File: Leetcode/tree/392_IsSubsequence/Solution.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if(s.length()==0)
            return true;
        Queue<Character> q = new LinkedList<>();
        for(Character c: s.toCharArray()){
            q.offer(c);
        }
        for(int i = 0; i<t.length(); ++i){
            if(!q.isEmpty() && t.charAt(i)==q.peek())
                q.poll();
        }
        return q.isEmpty();
    }
}

File: Leetcode/tree/392_IsSubsequence/Solution.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if s=="":
            return True
        l1 = list(s)
        l2 = list(t)
        for i in range(len(l2)):
            if l1 and l1[0] == l2[i]:
                l1.pop(0)
        return l1 == []
